flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/AbstractRocksDBState.java

```diff
@@ -154,6 +159,29 @@ public void setCurrentNamespace(N namespace) {
 		return backend.db.get(columnFamily, tmpKeySerializationView.getCopyOfBuffer());
 	}
 
+	public byte[] migrateSerializedValue(
```

Comment:
I think this method does not belong here. This class is used to interact with the state during event processing, I suggest to keep that separated from migration concerns. If you are looking for a way to differentiate the logic for different states, you could also just use a factor that gives you different implementations of some `StateValueTransformer`.

Reply:
Any thoughs about this or was it overlooked?